eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -221,8 +221,8 @@ class Channel(val nodeParams: NodeParams, val wallet: EclairWallet, remoteNodeId
             case None =>
               // in all other cases we need to be ready for any type of closing
               // TODO: should we wait for an acknowledgment from the watcher?
-              blockchain ! WatchSpent(self, data.commitments.commitInput.outPoint.txid, data.commitments.commitInput.outPoint.index.toInt, data.commitments.commitInput.txOut.publicKeyScript, BITCOIN_FUNDING_SPENT)
-              blockchain ! WatchLost(self, data.commitments.commitInput.outPoint.txid, nodeParams.minDepthBlocks, BITCOIN_FUNDING_LOST)
+              blockchain ! WatchSpent(self, data.commitments.fundingTxId, data.commitments.commitInput.outPoint.index.toInt, data.commitments.commitInput.txOut.publicKeyScript, BITCOIN_FUNDING_SPENT)
+              blockchain ! WatchLost(self, data.commitments.fundingTxId, nodeParams.minDepthBlocks, BITCOIN_FUNDING_LOST)
```

Comment:
Why did you change that? I think the previous code was less confusing since we still need to refer to `data.commitments.commitInput` in the other parameters...and same for the other places in this file where that code was changed.

Reply:
Originally it's because of this place: https://github.com/ACINQ/eclair/pull/1542/files#diff-86ff21f0015ed1b8252f65273dd4d06aR108 It accepts `ChannelCommitments` trait but then tries to access a funding txid which `ChannelCommitments` trait does not have (because HC has no on-chain funding). The plan to set it to all zeros in HC data. Then I've decided to apply it elsewhere since it's already defined.

Reply:
Maybe another way to do it is to add something more general, like "fundingDescription" field which would resolve to txid in normal channel and, say, human readable "No on-chain funding" in HC?

Reply:
Honestly the GUI is just a demo tool, if that's the only place where this is used, don't bother, it should not make eclair-core more complex